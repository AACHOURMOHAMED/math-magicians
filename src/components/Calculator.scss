.Hero {
  margin-top: 150px;

  .buttons {
    position: relative;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: center;

    .text {
      width: 100%;
      height: 70px;
      padding: 15px 10px;
      font-size: 25px;
      background-color: rgb(99, 92, 92);
      border: none;
      color: #fff;
      text-align: right;
    }

    span {
      font-size: 25px;
      position: absolute;
      top: 2%;
      right: 5%;
      color: #ffff;
    }
  }
}
