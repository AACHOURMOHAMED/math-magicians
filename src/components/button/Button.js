/* eslint-disable react/prop-types, react/prefer-stateless-function */

import React, { Component } from 'react';
import './button.scss';

export class button extends Component {
  constructor(props) {
    super(props);
    this.handleButtonClick = this.handleButtonClick.bind(this);
  }

  // eslint-disable-next-line class-methods-use-this
  handleButtonClick(e) {
    const { handleButtonClick } = this.props;
    handleButtonClick(e.target.textContent);
  }

  render() {
    const { children, buttonStyle } = this.props;
    const STYLES = ['btn--primary', 'btn--sec', 'btn--zero'];
    const CheckBtnStyles = STYLES.includes(buttonStyle) ? buttonStyle : STYLES[0];
    return (
      <button className={`btn ${CheckBtnStyles}`} onClick={this.handleButtonClick} type="button">{children}</button>
    );
  }
}

export default button;
