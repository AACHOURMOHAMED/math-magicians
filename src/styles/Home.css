.homepage {
  width: 80%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 50px auto;
}

h2 {
  font-size: 2rem;
  font-weight: bold;
  color: #000;
  margin-bottom: 1rem;
}

p {
  font-size: 1.5rem;
  color: #000;
  margin-bottom: 1rem;
}
