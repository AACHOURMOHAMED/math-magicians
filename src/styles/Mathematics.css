.calculator-page {
  display: flex;
  max-width: 90%;
  align-items: center;
  margin: 0 auto;
}

.calculator-content {
  flex: 1;
  padding: 0 2.5vw;
  font-size: 2rem;
}
