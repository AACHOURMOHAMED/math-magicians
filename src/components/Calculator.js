import React, { useState } from 'react';
import Button from './button/Button';
import './Calculator.scss';
import Result from './result';
import calculate from '../logic/calculate';

let obj = {};

const Calculator = () => {
  const [output, setOutput] = useState(0);

  const handleButtonClick = (buttonName) => {
    obj = calculate(obj, buttonName);

    let outStr = '';
    if (obj.total) {
      outStr += obj.total;
    }
    if (obj.operation) {
      outStr += obj.operation;
    }
    if (obj.next) {
      outStr += obj.next;
    }
    if (outStr === '') {
      outStr += '0';
    }

    setOutput(outStr);
  };
  return (
    <main className="Hero">
      <div className="container">
        <div className="buttons">
          <Result output={output} />
          <Button handleButtonClick={handleButtonClick}>AC</Button>
          <Button handleButtonClick={handleButtonClick}>+/-</Button>
          <Button handleButtonClick={handleButtonClick}>%</Button>
          <Button buttonStyle="btn--sec" handleButtonClick={handleButtonClick}>รท</Button>
          <Button handleButtonClick={handleButtonClick}>7</Button>
          <Button handleButtonClick={handleButtonClick}>8</Button>
          <Button handleButtonClick={handleButtonClick}>9</Button>
          <Button buttonStyle="btn--sec" handleButtonClick={handleButtonClick}>x</Button>
          <Button handleButtonClick={handleButtonClick}>4</Button>
          <Button handleButtonClick={handleButtonClick}>5</Button>
          <Button handleButtonClick={handleButtonClick}>6</Button>
          <Button buttonStyle="btn--sec" handleButtonClick={handleButtonClick}>-</Button>
          <Button handleButtonClick={handleButtonClick}>1</Button>
          <Button handleButtonClick={handleButtonClick}>2</Button>
          <Button handleButtonClick={handleButtonClick}>3</Button>
          <Button buttonStyle="btn--sec" handleButtonClick={handleButtonClick}>+</Button>
          <Button buttonStyle="btn--zero" handleButtonClick={handleButtonClick}>0</Button>
          <Button handleButtonClick={handleButtonClick}>.</Button>
          <Button buttonStyle="btn--sec" handleButtonClick={handleButtonClick}>=</Button>
        </div>
      </div>
    </main>
  );
};

export default Calculator;
