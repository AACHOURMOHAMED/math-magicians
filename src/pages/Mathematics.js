import React from 'react';
import '../styles/Mathematics.css';
import Calculator from '../components/Calculator';

function Mathematics() {
  return (
    <section className="calculator-page">
      <div className="calculator-content">
        <h2>Let&apos;s do some math!</h2>
      </div>
      <Calculator />
    </section>
  );
}

export default Mathematics;
