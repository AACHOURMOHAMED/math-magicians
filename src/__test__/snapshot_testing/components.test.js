import TestRenderer from 'react-test-renderer';
import Calculator from '../../components/Calculator';
import Button from '../../components/Button/Button';
import Home from '../../pages/Home';
import Mathematics from '../../pages/Mathematics';
import Quotes from '../../pages/Quotes';

describe('test screenshots for all components', () => {
  test('matches Calculator snapshot', () => {
    const tree = TestRenderer.create(<Calculator />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Button snapshot', () => {
    const tree = TestRenderer.create(<Button />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Home snapshot', () => {
    const tree = TestRenderer.create(<Home />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Mathematics snapshot', () => {
    const tree = TestRenderer.create(<Mathematics />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Quote snapshot', () => {
    const tree = TestRenderer.create(<Quotes />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
