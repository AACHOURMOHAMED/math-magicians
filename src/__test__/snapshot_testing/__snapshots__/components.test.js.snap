// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test screenshots for all components matches Button snapshot 1`] = `
<button
  className="btn btn--primary"
  onClick={[Function]}
  type="button"
/>
`;

exports[`test screenshots for all components matches Calculator snapshot 1`] = `
<main
  className="Hero"
>
  <div
    className="container"
  >
    <div
      className="buttons"
    >
      <div
        className="text"
      >
        0
      </div>
      <button
        className="btn btn--primary"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn btn--primary"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn btn--primary"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn btn--sec"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn btn--primary"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn btn--primary"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn btn--primary"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn btn--sec"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn btn--primary"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn btn--primary"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn btn--primary"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn btn--sec"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn btn--primary"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn btn--primary"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn btn--primary"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn btn--sec"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn btn--zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn btn--primary"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn btn--sec"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</main>
`;

exports[`test screenshots for all components matches Home snapshot 1`] = `
<section
  className="homepage"
>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit, quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos sapiente officiis modi at sunt excepturi expedita sint? Sed quibusdam recusandae alias error harum maxime adipisci amet laborum.
    <br />
    <br />
    Minima nesciunt dolorem! Officiis iure rerum voluptates a cumque velit quibusdam sed amet tempora. Sit laborum ab, eius fugit doloribus tenetur fugiat, temporibus enim commodi iusto libero magni deleniti quod quam consequuntur! Commodi minima excepturi repudiandae velit hic maxime doloremque. Quaerat provident commodi consectetur veniam similique ad earum omnis ipsum saepe, voluptas, hic voluptates pariatur est explicabo fugiat, dolorum eligendi quam cupiditate excepturi mollitia maiores labore suscipit quas? Nulla, placeat. Voluptatem quaerat non architecto ab laudantium modi minima sunt esse temporibus sint culpa, recusandae aliquam numquam totam ratione voluptas quod exercitationem fuga. Possimus quis earum veniam quasi aliquam eligendi, placeat qui corporis!
  </p>
</section>
`;

exports[`test screenshots for all components matches Mathematics snapshot 1`] = `
<section
  className="calculator-page"
>
  <div
    className="calculator-content"
  >
    <h2>
      Let's do some math!
    </h2>
  </div>
  <main
    className="Hero"
  >
    <div
      className="container"
    >
      <div
        className="buttons"
      >
        <div
          className="text"
        >
          0
        </div>
        <button
          className="btn btn--primary"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="btn btn--primary"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="btn btn--primary"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="btn btn--sec"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          className="btn btn--primary"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="btn btn--primary"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="btn btn--primary"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="btn btn--sec"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          className="btn btn--primary"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="btn btn--primary"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="btn btn--primary"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="btn btn--sec"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className="btn btn--primary"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="btn btn--primary"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="btn btn--primary"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="btn btn--sec"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="btn btn--zero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="btn btn--primary"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="btn btn--sec"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </main>
</section>
`;

exports[`test screenshots for all components matches Quote snapshot 1`] = `
<section
  className="quote-page"
>
  <p />
  <h3>
    - 
  </h3>
</section>
`;
